src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -18,192 +18,181 @@ internal sealed class TaskCenterSolutionAnalysisProgressReporter
     {
         private static readonly TimeSpan s_minimumInterval = TimeSpan.FromMilliseconds(200);
 
+        #region Fields Protected By Lock
+
+        /// <summary>
+        /// Gate access to reporting sln crawler events so we cannot
+        /// report UI changes concurrently.
+        /// </summary>
+        private readonly object _lock = new object();
+
+        /// <summary>
+        /// Task used to trigger throttled UI updates in an interval
+        /// defined by <see cref="s_minimumInterval"/>
+        /// Protected from concurrent access by the <see cref="_lock"/>
+        /// </summary>
+        private Task? _intervalTask;
```

Comment:
i don't think tihs is correct.  your interval task isn't protected by _lock.  it's protected by only being accessed by the serialized _updateUITask

Reply:
n/m!